crates/swc_html_parser/src/parser/mod.rs

```diff
@@ -1861,13 +1889,18 @@ where
                     //
                     // Push "in template" onto the stack of template insertion modes so that it is
                     // the new current template insertion mode.
-                    Token::StartTag { tag_name, .. } if tag_name == "template" => {
+                    Token::StartTag {
+                        tag_name,
+                        is_self_closing,
+                        ..
+                    } if tag_name == "template" => {
                         self.insert_html_element(token_and_info)?;
                         self.active_formatting_elements.insert_marker();
                         self.frameset_ok = false;
                         self.insertion_mode = InsertionMode::InTemplate;
                         self.template_insertion_mode_stack
                             .push(InsertionMode::InTemplate);
+                        maybe_allow_self_closing!(is_self_closing, tag_name);
```

Comment:
Same here, so let's avoid do it there it unnecessary for perf reasons and avoid complex code where it unnecessary

Reply:
Found only a single test with this usage.
https://github.com/vuejs/core/blob/a41c5f1f4367a9f41bcdb8c4e02f54b2378e577d/packages/compiler-sfc/__tests__/parse.spec.ts#L196

In general I was trying to match the `Vue's template parser respects /> as an indication to end any tag, regardless of its type` premise. It is pretty convenient from a developer standpoint to not think about what can and what cannot be self-closed at all.

Regarding performance, I'd expect `rustc` to drop this branch altogether after static analysis (because `allow_self_closing` is set to `false` everywhere and condition always evaluates to `false`). But it's just a wild guess.

Performance is partially why I suggested making it a feature flag. But because features are merged, it is probably not a good idea.

Let's measure and see.